x/fswap/keeper/proposal.go

```diff
@@ -43,6 +44,8 @@ func (k Keeper) MakeSwap(ctx sdk.Context, swap types.Swap) error {
 	if err != nil {
 		return err
 	}
+
+	k.SetDenomMetaData(ctx, toDenomMetadata)
```

Comment:
We may prevent modifying any existing denom metadata.

Reply:
What conditions can be used?
Should we allow `SetDenomMetaData` only for new `Swap`?

Reply:
There would be a possibility to override existing denom for a new swap (e.g. set `to_denom` to cony in finschia application). You may get denom metadata of the proposal, and if it already exists, compare the values. It should fail if the denom metadata differs from the existing one.

Reply:
Should we prevent existing metadata when it is different to new one?

Reply:
It depends. If we want to open a way to modify existing metadatas, it would not be required. If we just want to "add" a new metadata, we have to compare the values.

Reply:
>How about accepting metadata only for the first time?
When I think about it again, there would be more swap with the same to-denom..